### PR TITLE
INT-8363 - Make Socket Mapping and Runtime Dynamic

### PR DESCRIPTION
**Pull request by koraytugay**
https://sonatype.atlassian.net/browse/INT-8363

### CODE REVIEW

File: src/main/java/com/neuvector/model/NVScanner.java [hunk 1]
```diff
@@ -38,14 +40,16 @@ public NVScanner(String nvScannerImage, String nvRegistryURL, String nvRegistryU
      * @param nvMountPath  The mount path mapping to the path inside the NeuVector Scanner container. It is the path to store the scan result. It is optional. If you don't pass in <code>nvMountPath</code>, it will use the default path "/var/neuvector"
      * @param bindMountShared Indicates whether the bind mount content needs to be shared. Using True when it's needed. If null is received, the default value, which is False, will be used.
      */
-    public NVScanner(String nvScannerImage, String nvRegistryURL, String nvRegistryUser, String nvRegistryPassword, String nvMountPath, Logger log, Boolean bindMountShared){
+    public NVScanner(String nvScannerImage, String nvRegistryURL, String nvRegistryUser, String nvRegistryPassword, String nvMountPath, Logger log, Boolean bindMountShared, String runtime, String socketMapping){
```
Comment (eduard-tita) [inline src/main/java/com/neuvector/model/NVScanner.java:43]: Can you add javadocs for the new params as well?

File: src/main/java/com/neuvector/model/NVScanner.java [hunk 2]
```diff
@@ -10,6 +10,8 @@ public class NVScanner {
     String nvMountPath;
     Logger log;
     Boolean bindMountShared = false;
+    String runtime;
+    String socketMapping;
```
Comment (eduard-tita) [inline src/main/java/com/neuvector/model/NVScanner.java:14]: I think it's safer to set this two new fields to their default values.

File: src/main/java/com/neuvector/model/NVScanner.java (hunk 1 above)
Comment (koraytugay) [reply src/main/java/com/neuvector/model/NVScanner.java:46]: Done in 232194fe45841f1e6ee155f0d3a86254cc1c8291 and c6511d28c303e030b3c334f7adc6621d972ca7e7

File: src/main/java/com/neuvector/model/NVScanner.java (hunk 2 above)
Comment (koraytugay) [reply src/main/java/com/neuvector/model/NVScanner.java:15]: Done in 232194fe45841f1e6ee155f0d3a86254cc1c8291 and c6511d28c303e030b3c334f7adc6621d972ca7e7

File: src/main/java/com/neuvector/DockerRunCommandBuilder.java [hunk 3]
```diff
@@ -10,9 +10,9 @@ public class DockerRunCommandBuilder
 {
     private final List<String> cmdList;
 
-    public DockerRunCommandBuilder() {
+    public DockerRunCommandBuilder(String runtime) {
```
[the remark this answers is not in the document]
Comment (koraytugay) [reply src/main/java/com/neuvector/DockerRunCommandBuilder.java:20]: Done in 232194fe45841f1e6ee155f0d3a86254cc1c8291 and c6511d28c303e030b3c334f7adc6621d972ca7e7

File: src/main/java/com/neuvector/DockerRunCommandBuilder.java [hunk 4]
```diff
@@ -11,8 +11,15 @@ public class DockerRunCommandBuilder
     private final List<String> cmdList;
 
     public DockerRunCommandBuilder() {
+        cmdList = new ArrayList<>("docker");
+        cmdList.add(runtime);
```
Comment (koraytugay) [inline src/main/java/com/neuvector/DockerRunCommandBuilder.java:15]:
```suggestion
```

File: src/main/java/com/neuvector/DockerRunCommandBuilder.java [hunk 5]
```diff
@@ -11,8 +11,14 @@ public class DockerRunCommandBuilder
     private final List<String> cmdList;
 
     public DockerRunCommandBuilder() {
+        cmdList = new ArrayList<>("docker");
+        cmdList.add("run");
```
Comment (koraytugay) [inline src/main/java/com/neuvector/DockerRunCommandBuilder.java:15]:
```suggestion
        cmdList = new ArrayList<>();
        cmdList.add("docker");
```

File: src/main/java/com/neuvector/model/NVScanner.java (hunk 2 above)
Comment (eduard-tita) [reply src/main/java/com/neuvector/model/NVScanner.java:15]: What I meant is this:
```suggestion
    String socketMapping = Scanner.SOCKET_MAPPING;
```
Same for `runtime`. If any constructor that doesn't set them is called, they will be `null`.

Comment (koraytugay) [reply src/main/java/com/neuvector/model/NVScanner.java]: Sorry, here: 999d9bace00dd0786b0562b7e01e3922b63ff834

**Review by eduard-tita — APPROVED** [review]
+1 and thank you!

**Comment by koraytugay** [comment]
@songlongcn Can we get this merged when time permits please? And a new release for this library so we can consume it?

**Comment by songlongcn** [comment]
Could you please update the README to reflect the changes in Socket Mapping and Runtime Dynamic ?

**Comment by koraytugay** [comment]
@songlongcn Done in b14aaba673b5cc8e92b6aec15ac397f3b3ff8b37

**Pull request merged**